File: Project Files/CarPark/car_park_JS_Start/src/CarPark.js
/* global Zone */
class CarPark { // eslint-disable-line no-unused-vars
  constructor () {
    this.allMyZones = []
  }

  addZone (newId, newLocation) {
    let newZone = new Zone(newId, newLocation)
    this.allMyZones.push(newZone)
  }

  findZone (targetId) {
    return this.allMyZones.find(aZone => aZone.id === targetId)
  }

  sortZones () {
    this.allMyZones.sort(function (a, b) {
      return a.id - b.id
    })
  }
  getZones () {
	// add code here
    }
    return result
  }
  getTemps () {
    let result = ''
    this.sortZones()
    for (let aZone of this.allMyZones) {
      result += aZone.getTemps() 
    }
    return result
  }
}

File: Project Files/CarPark/car_park_python_Answer/carpark.py
from zone import Zone


class CarPark:
    def __init__(self):
        self.all_my_zones = []

    def add_zone(self, new_id, new_where):
        new_zone = Zone(new_id, new_where, self)
        self.all_my_zones.append(new_zone)

    def find_zone(self, target_zone_id):
        return next((zone for zone in self.all_my_zones if zone.id == target_zone_id), None)

    def sort_zones(self):
        self.all_my_zones = sorted(self.all_my_zones, key=lambda zone: zone.id)

    def get_zones(self):
        self.sort_zones()
        out = ''
        for a_zone in self.all_my_zones:
            out += str(a_zone) + '\n'
        return out

    def get_those_with_two_clients(self):
        self.sort_zones()
        out = ''
        for a_zone in self.all_my_zones:
            if a_zone.has_two_clients():
                out += str(a_zone) + '\n'
            out += a_zone.get_client()
        return out
